server/routes/users.js

```diff
@@ -424,30 +413,30 @@ router.post('/verify', async ctx => {
  * @apiParam (Required Params) {string} user[email] Unique email
  *
  */
-router.get('/verify', async ctx => {
+
+router.get('/verify', permController.requireAuth, async ctx => {
   const decodedMail = Buffer.from(ctx.query.email, 'base64').toString('ascii');
-  const user = await User.query().findOne({ 'email': decodedMail, 'resetPasswordToken': ctx.query.token });
-  ctx.assert(user, 'No email found');
-  let message;
+  let user = await User.query().findOne({ 'email': decodedMail });
```

Comment:
someone can easily bypass verification by making an API request to this endpoint with email query parameter e.g. `/api/v1/users/${randomId}/verify?email=${base64EncodedEmail}`

the endpoint should verify that the token received is associated to the user email first.